components/engine_rocks/src/engine.rs

```diff
@@ -51,15 +52,15 @@ impl RocksEngine {
 
 impl KvEngine for RocksEngine {
     type Snapshot = RocksSnapshot;
-    type WriteBatch = crate::WriteBatch;
+    type WriteBatch = RocksWriteBatch;
 
-    fn write_opt(&self, opts: &WriteOptions, wb: &Self::WriteBatch) -> Result<()> {
+    fn write_opt(&self, wb: &Self::WriteBatch, opts: &WriteOptions) -> Result<()> {
         if wb.get_db().path() != self.0.path() {
```

Comment:
Is this compare necessary for every write operations?

Reply:
Seems odd to me. I'll turn it into a debug assert.